src/preamble.js

```diff
@@ -1164,6 +1164,16 @@ assert(typeof Int32Array !== 'undefined' && typeof Float64Array !== 'undefined'
 #endif
 
 #if IN_TEST_HARNESS
+
+// Test runs in browsers should always be free from uncaught exceptions. If an uncaught exception is thrown, we fail browser test execution in the REPORT_RESULT() macro to output an error value.
+if (typeof window !== 'undefined') {
+  window.addEventListener('error', function(e) {
+    if (e == 'SimulateInfiniteLoop' || e.message.indexOf('SimulateInfiniteLoop') != -1) return;
```

Comment:
> just e === 'SimulateInfiniteLoop' should be enough

That alone did not work. According to [MDN on onerror](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror), the parameter `e` can either be a string or an error object, and in the test runner on the updated `test_sdl_touch` test, it is the latter and `e.message.indexOf('SimulateInfiniteLoop') != -1` was needed.